### PR TITLE
clientv3/concurrency: Document Election API context parameter

### LINKED ISSUE #9377: Document Election API context parameter
Just like https://godoc.org/github.com/coreos/etcd/clientv3#Watcher, document https://godoc.org/github.com/coreos/etcd/clientv3/concurrency#Election.Campaign

### PR DESCRIPTION
**Pull request by vimalk78**
fixes #9377

Please read https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by vimalk78** [comment]
9999th PR :)

File: clientv3/concurrency/election.go [hunk 1]
```diff
@@ -55,8 +55,24 @@ func ResumeElection(s *Session, pfx string, leaderKey string, leaderRev int64) *
...
+// key.
+// Multiple Sessions can participate in the election for the
+// same prefix, but only one can be the leader at a time.
+// A Session succeeds in election if it can create a key with the
+// prefix before other participating Sessions. If there are other
+// keys created before with the same prefix, the session waits for
+// previous keys to be deleted by doing a Watch() for the Delete events
+// on the previous keys with the same prefix.
+//
+// If the context is 'context.TODO/Background', the Compaign will continue
```
Comment (spzala) [inline clientv3/concurrency/election.go]: typo `Compaign` ...  @gyuho can verify the content at best, but couple of small comments here.

File: clientv3/concurrency/election.go [hunk 2]
```diff
@@ -55,8 +55,24 @@ func ResumeElection(s *Session, pfx string, leaderKey string, leaderRev int64) *
 	}
 }
 
-// Campaign puts a value as eligible for the election. It blocks until
-// it is elected, an error occurs, or the context is cancelled.
+// Campaign puts a value as eligible for the election on the prefix
+// key.
+// Multiple Sessions can participate in the election for the
+// same prefix, but only one can be the leader at a time.
+// A Session succeeds in election if it can create a key with the
```
Comment (spzala) [inline clientv3/concurrency/election.go]: "session" with small s instead of S.. here and other places?

File: clientv3/concurrency/election.go [hunk 3]
```diff
@@ -55,8 +55,24 @@ func ResumeElection(s *Session, pfx string, leaderKey string, leaderRev int64) *
...
+// previous keys to be deleted by doing a Watch() for the Delete events
+// on the previous keys with the same prefix.
+//
+// If the context is 'context.TODO/Background', the Compaign will continue
+// to be blocked for other keys to be deleted, unless server returns a non-recoverable
+// error (e.g. ErrCompacted).
+// Otherwise, until the conext is not cancelled or timed-out, Campaign will continue
+// to be blocked until it becomes the leader.
+//
+// Campaign blocks until the session is elected, an error occurs,
```
Comment (spzala) [inline clientv3/concurrency/election.go]: The last two original lines seems now taken care above so looks redundant.

**Comment by wenjiaswe** [comment]
cc @jingyih

**Comment by vimalk78** [comment]
request @gyuho to review this.
not sure why the CI keeps failing for this PR, this only contains comments change.
i rebased this commit to 484f62203410fcb1d6438ae42952eac2dc7f4ceb

**Comment by vimalk78** [comment]
@gyuho , pls check this?

File: clientv3/concurrency/election.go [hunk 4]
```diff
@@ -55,8 +55,21 @@ func ResumeElection(s *Session, pfx string, leaderKey string, leaderRev int64) *
 	}
 }
 
-// Campaign puts a value as eligible for the election. It blocks until
-// it is elected, an error occurs, or the context is cancelled.
+// Campaign puts a value as eligible for the election on the prefix
+// key.
+// Multiple sessions can participate in the election for the
+// same prefix, but only one can be the leader at a time.
+// A session succeeds in election if it can create a key with the
```
Comment (xiang90) [inline clientv3/concurrency/election.go]: 62-66 are implementation details that we want to hide from the users. simply telling users what to expect is good enough.

**Comment by xiang90** [comment]
lgtm after hiding the implementation details.

**Comment by xiang90** [comment]
lgtm. @gyuho can you check why CI failed?

**Comment by gyuho** [comment]
@xiang90 @vimalk78 Tests have been flaky. LGTM. Thanks!

**Pull request merged**